File: controle/controle_reserva.php
<?php
require_once '../dao/dao_reserva.php';
require_once '../model/reserva.php';
require_once '../dao/dao_livro.php';
require_once '../model/livro.php';
require_once '../dao/dao_cliente.php';
require_once '../model/cliente.php';

session_start();

$op = $_POST["op"];

$reserva = new Reserva();
$reserva->setAtivo(true);
$reserva->setData_reserva($_POST["data_reserva"]);
$reserva->setData_retirada($_POST["data_retirada"]);
// $reserva->setId_cliente($_POST["id_cliente"]);
// $reserva->setId_livro($_POST["id_livro"]);

echo $reserva;

$daoCliente = new DaoCliente();
$daoLivro   = new DaoLivro();
$daoReserva = new DaoReserva();

try {
    if ($op == "salvar") {

        $cliente = $daoCliente->buscar_por_login($_POST["id_cliente"]);
        if ($cliente != null) {
            $reserva->setId_cliente($cliente);
        } else {
            echo "Cliente Errado!";
            $op = "falha";
        }

        $livro = $daoLivro->busca_por_codigo($_POST["id_livro"]);
        if ($livro != null) {
            $reserva->setId_livro($livro);
        } else {
            echo "Livro Invalido!";
            $op = "falha";
        }

        if ($op != "falha") {

            $r = $daoReserva->salvar($reserva);

            if ($r == true) {
                echo "Sucesso";
            } else {
                echo "Falha";
            }
        }

    } else if ($op == "editar") {
        $daoReserva->editar($reserva);
        echo "Sucesso";

    } else if ($op == "buscabusca") {
        $reservas = $daoReserva->busca_por_busca($reserva);
        echo json_encode($reservas);

    } else if ($op == "buscaid") {
        $reservas = $daoReserva->buscar_por_id($_SESSION["id"]);
        ob_end_clean(); //limpo conteudo anterior (apago buffer)
        ob_start(); //inicializo novamente o buffer
        echo json_encode($reservas);

    } else if ($op == "remover") {
        $reserva->setId_cliente($_POST["id_cliente"]);
        $reserva->setId_livro($_POST["id_livro"]);
        $daoReserva->remover($reserva);
        echo "Sucesso";
    }
} catch (Throwable $th) {
    echo "Falha" . $th;
}

File: controle/ControleReserva.py
from flask import render_template, session, redirect, url_for, request, json, Blueprint
from dao.DaoLivro import DaoLivro
from dao.DaoCliente import DaoCliente
from dao.DaoReserva import DaoReserva
from model.Reserva import Reserva

reserva = Blueprint("reserva", "Biblioteca", template_folder="../view", static_folder="../estilo")
daoLivro = DaoLivro()
daoCliente = DaoCliente()
daoReserva = DaoReserva()

@reserva.route("/Livro/Reserva/<codigo>/<login>")
def reserva_entrar(codigo, login):
    if 'logado' in session:

        cliente = daoCliente.search_login(login)

        template = render_template("cabecalho.html")
        template += render_template("logado.html")
        template += render_template("reserva.html", codigo=codigo, cliente=cliente.cpf)
        template += render_template("rodape.html")
        return template
    else:
        return redirect(url_for('entrada.index'))

@reserva.route("/Livro/Reservar/", methods=['POST'])
def reservar():
    try:
        data_reserva = request.form['data_reserva'];
        data_retirada = request.form['data_retirada'];
        id_livro = request.form['id_livro'];
        id_cliente = request.form['id_cliente'];

        clienteM = daoCliente.search_cpf(id_cliente)
        livroM = daoLivro.search_codigo(id_livro)

        reservaM = Reserva(data_reserva=data_reserva, data_retirada=data_retirada, id_livro=livroM.id, id_cliente=clienteM.id)
        daoReserva.create(reservaM)
        return json.dumps({'status': 'OK'});
    except Exception as e:
        return json.dumps({'status': 'Erro'});

@reserva.route("/Livro/Reserva/Buscar/", methods=['POST'])
def buscar_reserva():
    try:
        reservas = daoReserva.search_id_cliente(session['logado'])

        j = []
        for r in reservas:
            j.append({
                'id' : r[0].id,
                'titulo' : r[2].titulo,
                'data_reserva' : r[0].data_reserva,
                'data_retirada' : r[0].data_retirada,
                'nome' : r[1].nome
            })
        return json.dumps(j)
    except Exception as e:
        return json.dumps({"status': 'Erro"})

@reserva.route("/Livro/Reserva/Cancelar/", methods=['POST'])
def cancelar_reserva():
    try:

        id = request.form['id'];
        reservaM = daoReserva.search_id(id)
        daoReserva.remove(reservaM)

        return json.dumps({"status": "Ok"})
    except Exception as e:
        return json.dumps({"status": "Erro"})
